addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -1833,24 +1844,36 @@ export function getRangePosition(el, document, options = {}) {
     return offset;
 }
 
-export const leftDeepFirstPath = createDOMPathGenerator(DIRECTIONS.LEFT, false, false);
-export const leftDeepOnlyPath = createDOMPathGenerator(DIRECTIONS.LEFT, true, false);
-export const leftDeepFirstInlinePath = createDOMPathGenerator(DIRECTIONS.LEFT, false, true);
-export const leftDeepOnlyInlinePath = createDOMPathGenerator(DIRECTIONS.LEFT, true, true);
-export const leftDeepOnlyInlineInScopePath = createDOMPathGenerator(
-    DIRECTIONS.LEFT,
-    true,
-    true,
-    true,
-);
-
-export const rightDeepFirstPath = createDOMPathGenerator(DIRECTIONS.RIGHT, false, false);
-export const rightDeepOnlyPath = createDOMPathGenerator(DIRECTIONS.RIGHT, true, false);
-export const rightDeepFirstInlinePath = createDOMPathGenerator(DIRECTIONS.RIGHT, false, true);
-export const rightDeepOnlyInlinePath = createDOMPathGenerator(DIRECTIONS.RIGHT, true, true);
-export const rightDeepOnlyInlineInScopePath = createDOMPathGenerator(
-    DIRECTIONS.RIGHT,
-    true,
-    true,
-    true,
-);
+const isNotEditableNode = node =>
+    node.getAttribute && node.getAttribute('contenteditable') === 'false';
+const isNotBlock = node => !isBlock(node);
+
+export const leftLeafFirstPath = createDOMPathGenerator(DIRECTIONS.LEFT);
+export const leftLeafOnlyNotBlockPath = createDOMPathGenerator(DIRECTIONS.LEFT, {
```

Comment:
I find these names actually harder to read than the lists of options. How often do we use these? Do we really need them or could we technically just call `createDOMPathGenerator` when we need them?
If we have to keep them, maybe a one-line comment for each would be good, to describe the generated path.

Reply:
Having a comment would make no sense to me, it would just make things more complicated.
Reading the attributes and the `createDOMPathGenerator` documentation is the easiest way to understand.

Reply:
You answered the second part of my message but it's the first part that mattered most. How often do we use these? Do we really need them?
Because as is, with these names, it's cumbersome and unclear.